core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -75,7 +75,12 @@ public CarbonTablePath(CarbonTableIdentifier carbonTableIdentifier, String table
    * @param carbonFilePath
    */
   public static String getFolderContainingFile(String carbonFilePath) {
-    return carbonFilePath.substring(0, carbonFilePath.lastIndexOf('/'));
+    int lastIndex = carbonFilePath.lastIndexOf('/');
+    // below code for handling windows environment
+    if (-1 == lastIndex) {
```

Comment:
Whether it need support this scenario: E:/xubo\idea ?

Reply:
in windows file seperator is "\\" , this is to handle the same

Reply:
in windows, you code will return E: if you input  E:/xubo\idea , not return E:/xubo

Reply:
E:/xubo\idea  will never come either  it will be E:/xubo/idea or  E:\xubo\idea ?

Reply:
ok, it's fine